tests/distributed_mode/test_single_label_classifier.py

```diff
@@ -0,0 +1,94 @@
+import pytest
+
+import syft as sy
+from syft.generic.string import String
+from syfertext.pipeline.single_label_classifier import AverageDocEncoder, SingleLabelClassifier
+from syfertext.local_pipeline import get_test_language_model
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+# Create a torch hook for PySyft
+hook = sy.TorchHook(torch)
+seed = 42
+
+# Set up local and remote workers
+me = hook.local_worker
+me.is_client_worker = False
+
+alice = sy.VirtualWorker(hook, id='alice')
+bob = sy.VirtualWorker(hook, id='bob')
+crypto_provider = sy.VirtualWorker(hook, id='crypto_provider')
+
+# Create toy classifier
+class Net(sy.Plan):
+    def __init__(self):
+        super(Net, self).__init__()
+        self.fc = nn.Linear(300, 2)
+
+    def forward(self, x):
+        logits = self.fc(x)
+        probs = F.relu(logits)
+        return probs, logits
+
+doc_encoder = AverageDocEncoder()
+labels = ["A", "B"]
+
+
+
+@pytest.mark.parametrize("text,expected", [("The quick brown fox", "B")])
+def test_single_label_inference(text, expected):
+    torch.manual_seed(seed)
+    classifier_name = "classifier"
+    nlp = get_test_language_model()
+
+    # Set up SingleLabelClassifier and add to pipeline
+    net = Net()
+    classifier = SingleLabelClassifier(
+        classifier=net, 
+        doc_encoder=doc_encoder,
+        encryption=None,
+        labels=labels
+    )  
+    nlp.add_pipe(classifier, name=classifier_name, access={'*'})
+    nlp.deploy(worker=bob)
```

Comment:
let's add another test where you test the classifier from a deployed nlp object. Currently even if you are doing `nlp.deploy`, you are still using the local `nlp` object. 
So maybe remove it from here and make another test where use the classifier stored at `bob`.

Reply:
got it, i'll get to this asap. in that case I'll correct the ones here to use the classifier at bob, then add local tests in tests/local_mode? 

the original issue was focused on remote inference, but since it looks like there's no local test at the moment i feel it would make sense to add it

Reply:
Thanks for the feedback @bicycleman15! I've updated the tests to reflect the corrections (did not send String to bob in the earlier version). Please see the latest commit.

I think I need some help understanding why the nlp object is local in this case, since once it's deployed, when I check its subpipeline_templates everything is located in bob.

Reply:
Also, once this is fixed, should I write a local version of this in tests/local_mode? The original issue only specified remote inference but there doesn't seem to be any tests on local inference at the moment.

Reply:
Yeah sure. 👍 .